extensions/arc/src/ui/dashboards/postgres/postgresOverviewPage.ts

```diff
@@ -296,19 +318,22 @@ export class PostgresOverviewPage extends DashboardPage {
 	}
 
 	private refreshNodes() {
-		const nodes = this._postgresModel.numNodes();
 		const endpoint: { ip?: string, port?: number } = this._postgresModel.endpoint();
 
-		const data: any[][] = [];
-		for (let i = 0; i < nodes; i++) {
-			data.push([
-				`${this._postgresModel.name()}-${i}`,
-				i === 0 ? loc.coordinatorEndpoint : loc.worker,
-				i === 0 ? `${endpoint.ip}:${endpoint.port}` :
-					`${this._postgresModel.name()}-${i}.${this._postgresModel.name()}-svc.${this._postgresModel.namespace()}.svc.cluster.local`]);
-		}
+		this.nodesTable!.data = this._postgresModel.pods()?.map((pod: V1Pod) => {
+			const name = pod.metadata?.name;
+			const role: PodRole | undefined = PostgresModel.getPodRole(pod);
+			const service = pod.metadata?.annotations?.['arcdata.microsoft.com/serviceHost'];
+			const internalDns = service ? `${name}.${service}` : '';
+
+			return [
+				name,
+				role ? PostgresModel.getPodRoleName(role) : '',
```

Comment:
I'd just have the function be able to take in undefined values and return '' in that case - makes it simpler to call.